src/lib/Libwin/accesinfo.c

```diff
@@ -181,25 +185,39 @@ secure_file(char *path, char *user, ACCESS_MASK mask)
 	ACL	*pdacl;
 	struct  stat sbuf;
 	SECURITY_INFORMATION	si = 0;
-	char	logb[LOG_BUF_SIZE] = {'\0' } ;
 	char	*gname = NULL;
 
-	if (path == NULL || *path == '\0')
+	if (path == NULL || *path == '\0') {
+		log_eventf(PBSEVENT_ERROR, 0, LOG_DEBUG, __func__, "invalid path");
 		return (0);
+	}
 
-	if (lstat(path, &sbuf) == -1)
+	if (lstat(path, &sbuf) == -1) {
+		log_eventf(PBSEVENT_ERROR, 0, LOG_DEBUG, __func__, "Path don't exists");
 		return (0);	/* ignore non-existent files! */
+	}
 
-	if (!has_privilege(SE_RESTORE_NAME))
-		ena_privilege(SE_RESTORE_NAME);
+	if (!has_privilege(SE_RESTORE_NAME)) {
+		if (ena_privilege(SE_RESTORE_NAME) == 0) {
+			log_eventf(PBSEVENT_ERROR, 0, LOG_DEBUG, __func__, "Unable to enable privilege: SE_RESTORE_NAME");
+		}
+	}
 
-	if (!has_privilege(SE_TAKE_OWNERSHIP_NAME))
-		ena_privilege(SE_TAKE_OWNERSHIP_NAME);
+	if (!has_privilege(SE_TAKE_OWNERSHIP_NAME)) {
+		if (ena_privilege(SE_TAKE_OWNERSHIP_NAME) == 0) {
+			log_eventf(PBSEVENT_ERROR, 0, LOG_DEBUG, __func__, "Unable to enable privilege: SE_TAKE_OWNERSHIP_NAME");
+		}
+	}
 
-	if (!has_privilege(SE_SECURITY_NAME))
-		ena_privilege(SE_SECURITY_NAME);
+	if (!has_privilege(SE_SECURITY_NAME)) {
+		if (ena_privilege(SE_SECURITY_NAME) == 0) {
+			log_eventf(PBSEVENT_ERROR, 0, LOG_DEBUG, __func__, "Unable to enable privilege: SE_SECURITY_NAME");
```

Comment:
Are these DEBUG only messages - we need error logs printed always

Reply:
you can use log_err(-1,